_posts/2019-11-08-method-references-the-unpolished-gem-of-java-8.markdown

```diff
@@ -199,9 +199,9 @@ dogs.stream()
 The two examples above are obviously not flawless.
 
 The first one requires changes in the type safety mechanisms.
- * How can we be sure what is the type of the Method Reference before calling its' method?
+ * How can we be sure what is the type of the Method Reference before calling its method?
  * If it is always the type of method's target Functional Interface, then how would it work in assignment statements?
- * If the type is not arbitrary, how can we know it has the negation method?
+ * If the type is not arbitrary, how can we know it has the negation method? 
```

Comment:
Isn't this why we have type inference? Maybe I'm not understanding the part of arbitrary, so let me know :)

Reply:
This refers to casting method reference types explicitly like:
```java
.filter(((Predicate<Dog>) veterinarian::isHealthy).negate())
```
Sometimes Java is not smart enough (;